test/unit-test/test_derivation.ml

```diff
@@ -161,6 +161,31 @@ let%expect_test "derivation" =
           Empty))))
   |}];
 
+  (*
+   * [X : number()] |-
+   * case 123 of
+   *   X when X
+   * end
+   *)
+  let ctx = Context.empty |> Context.add (Context.Key.Var "X") (Type.of_elem TyNumber) in
+  print ctx (Case
+    (Constant (Number 123),
+      [(PatVar "X", Constant (Atom "true")), Var "X"]));
+  [%expect {|
+    (Ok (
```

Comment:
Reading memo:
With this PR, the code using a already defined variable in a case pattern can be correctly derived